src/ui/zcl_abapgit_gui_page_merge.clas.abap

```diff
@@ -184,11 +184,23 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_MERGE IMPLEMENTATION.
           zcx_abapgit_exception=>raise( 'nothing to merge' ).
         ENDIF.
 
-        CREATE OBJECT ei_page TYPE zcl_abapgit_gui_page_commit
-          EXPORTING
-            io_repo  = mo_repo
-            io_stage = mo_merge->get_result( )-stage.
-        ev_state = zif_abapgit_definitions=>gc_event_state-new_page.
+        IF mo_repo->get_local_settings( )-code_inspector_check_variant IS NOT INITIAL.
+
+          CREATE OBJECT ei_page TYPE zcl_abapgit_gui_page_code_insp
+            EXPORTING
+              io_repo  = mo_repo
+              io_stage = mo_merge->get_result( )-stage.
+          ev_state = zif_abapgit_definitions=>gc_event_state-new_page.
+
+        ELSE.
+
+          CREATE OBJECT ei_page TYPE zcl_abapgit_gui_page_commit
+            EXPORTING
+              io_repo  = mo_repo
+              io_stage = mo_merge->get_result( )-stage.
+          ev_state = zif_abapgit_definitions=>gc_event_state-new_page.
```

Comment:
line `ev_state = zif_abapgit_definitions=>gc_event_state-new_page.` is the same in both branches of the IF => can be moved to after ENDIF